packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -419,7 +422,34 @@ export class StripeHandler {
       promotionCode,
       priceId,
     });
-    return stripeInvoiceToInvoicePreviewDTO(previewInvoice);
+    return stripeInvoiceToFirstInvoicePreviewDTO(previewInvoice);
+  }
+
+  /**
+   * Preview invoices for an array of subscriptionIds
+   */
+  async subsequentInvoicePreview(
+    request: AuthRequest
+  ): Promise<invoiceDTO.subsequentInvoicePreviewSchema> {
+    this.log.begin('subscriptions.subsequentInvoicePreview', request);
+    await this.customs.checkIpOnly(request, 'subsequentInvoicePreview');
+
+    const { subscriptionIds } = request.payload as Record<
+      string,
+      Array<string>
+    >;
+
+    const subsequentInvoicePreviews = await Promise.all(
+      subscriptionIds.map((sub) =>
+        this.stripeHelper.previewInvoiceBySubscriptionId({
+          subscriptionId: sub,
+        })
+      )
+    );
+
+    return stripeInvoicesToSubsequentInvoicePreviewsDTO(
+      subsequentInvoicePreviews
+    );
```

Comment:
@chenba good spot! I've added some additional logic that should check this. Could you please take another look?

Reply:
Yeah, the additional check should work.

At this point though, is there a reason not to simply return the invoice previews for all of the customer's subscriptions?  So instead of accepting and verifying ids in the payload, why not not accept ids at all and use the subscriptions that belong to the customer?

Reply:
That makes sense. I made the necessary changes.

At the moment if a customer doesn't have any subscriptions, the endpoint will return a blank array `[]`. Do you think that's the correct behavior?

Reply:
Yes, an empty list make sense here.